runtime/j9vm/jvm.c

```diff
@@ -452,6 +456,74 @@ static const J9SignalMapping signalMap[] = {
 	{NULL, J9_SIG_ERR}
 };
 
+static void
+captureCommandLine(void)
+{
+#define ENV_VAR_NAME "IBM_JAVA_COMMAND_LINE"
+
+#if defined(LINUX)
+	int fd = open("/proc/self/cmdline", O_RDONLY, 0);
+
+	if (fd >= 0) {
+		char *buffer = NULL;
+		size_t length = 0;
+		for (;;) {
+			char small_buffer[512];
+			ssize_t count = read(fd, small_buffer, sizeof(small_buffer));
+			if (count <= 0) {
+				break;
+			}
+			length += (size_t)count;
+		}
+		if (length < 2) {
+			goto done;
+		}
+		/* final NUL is already included in length */
+		buffer = malloc(length);
+		if (NULL == buffer) {
+			goto done;
+		}
+		if ((off_t)-1 == lseek(fd, 0, SEEK_SET)) {
+			goto done;
+		}
+		if (read(fd, buffer, length) != length) {
+			goto done;
+		}
+		/* replace the internal NULs with spaces */
+		for (length -= 2;; length -= 1) {
+			if (0 == length) {
+				break;
+			}
+			if ('\0' == buffer[length]) {
+				buffer[length] = ' ';
+			}
+		}
+		/* it's not fatal if setenv() fails, so don't bother checking */
+		setenv(ENV_VAR_NAME, buffer, 0 /* don't overwrite */);
+		free(buffer);
+done:
```

Comment:
There is a memory leak error for some of the error cases that goto done.

Reply:
Good catch.

Reply:
Fixed in 229e47cda1681285905983db3863a8c897ba9eab.